cassandra-diagnostics-connector30/src/integration-test/java/io/smartcat/cassandra/diagnostics/connector/ITConnector.java

```diff
@@ -22,27 +23,32 @@
 
     private static Cluster cluster;
     private static Session session;
-    private static CountDownLatch lock = new CountDownLatch(1);
-    private static boolean queryIntercepted;
+    private static CountDownLatch lock = new CountDownLatch(2);
+    private static int queryInterceptedCnt;
 
     @BeforeClass
     public static void setUp() throws ConfigurationException, TTransportException, IOException, InterruptedException {
-        queryIntercepted = false;
+        queryInterceptedCnt = 0;
+
         final Instrumentation inst = InstrumentationSavingAgent.getInstrumentation();
         ConnectorConfiguration configuration = new ConnectorConfiguration();
+        configuration.enableTracing = true;
         final Connector connector = new ConnectorImpl();
         connector.init(inst, new QueryReporter() {
             @Override
             public void report(Query query) {
                 if (Query.StatementType.SELECT.equals(query.statementType()) && "test_keyspace"
-                        .equalsIgnoreCase(query.keyspace()) && "test_table".equalsIgnoreCase(query.tableName())) {
-                    queryIntercepted = true;
+                        .equalsIgnoreCase(query.keyspace()) && "test_table".equalsIgnoreCase(query.tableName()) &&
+                        query.statement().equalsIgnoreCase("SELECT uid FROM test_keyspace.test_table")) {
+                    queryInterceptedCnt++;
                     lock.countDown();
                 }
             }
         }, configuration);
+        
         EmbeddedCassandraServerHelper.startEmbeddedCassandra();
         connector.waitForSetupCompleted();
+        Thread.sleep(4000);
```

Comment:
Not yet. Too much time spent determining when Cassandra is really really ready to accept statements from client.